### ISSUE TITLE
Need the more complete code of AnyDA

### ISSUE THREAD

**Issue by dunanyang** [omchakrabarty/omchakrabarty.github.io#1]
Excuse me!
Thank for your wonderful work of AnyDA.
When I reproduce your results shown in your paper, I found a gap of three points. Therefore, I need a more completed code. I have tried many times to contact you. I'm looking forward your reply.
I will delete this issue after receiving your reply.
Thank you very much.

**Comment by omchakrabarty** [comment]
Hi @dunanyang! Thanks for reaching out. Can you be more specific as to in which dataset and which task you are facing this issue? The gap may be due to the particular seed or split of a certain domain in a specific dataset.

**Comment by dunanyang** [comment]
Hello @omchakrabarty! Thank you very much for taking the time to reply to my message. I am a second year graduate student majoring in Computer Technology from Tsinghua University. As I am particularly interested in your work, I would like to disturb you. My question may be as follows:
1. There is no warmup part of the code, so I don't know what precision can be used for training.
2. I saw errors in the pseudo label loss section and issues with the DomainSpecificBatchNormal section of the code.
The adversarial loss section is declared in the code but not implemented.
3. For the hyperparameters of the experiment, it is mentioned in the paper, and I need a more detailed one.
The above is my question. Because I am very interested in your work, and I am very grateful for your outstanding contributions in this field. May I ask if you could send a more complete version of the code? So that I can better learn about your work. Thank you, I look forward to your reply. My email is dny22@mails.tsinghua.edu.cn. Thank you very much!

**Comment by dunanyang** [comment]
Especially when I adapt from small datasets to  large datasets, such as Dlsy to Amazon, the results differ by 6 percentage points from your results in the paper. So, I particularly need your more complete code to reproduce your results.

**Comment by omchakrabarty** [comment]
@dunanyang, as I see that you are telling about the Office-31 dataset and the task of D -> A. I am not sure if you are using the source-pretrained model correctly. If you go through the paper, for the source pretraining you can use the same code but only use the student network. Simply train the student network using just the CE loss on the source data. Use it to initialise both the student and the teacher networks while starting the AnyDA training. For the source-only training, we obtained the accuracies for Office-31 as follows:
<img width="1012" alt="Sonly_Vals" src="https://github.com/omchakrabarty/omchakrabarty.github.io/assets/20840734/099d6f8c-ff22-4ef0-8dc0-df726c2cec4b">

Following this you can do the AnyDA training and you should get the results close. For any further queries, kindly drop a mail at **omchakrabarty@gmail.com**

**Comment by dunanyang** [comment]
Thank you very much. As I did not receive a response from you when I sent you my email earlier, I would like to reiterate my doubts below. For example, in the A2W task in Office31, I obtained the warmup result as shown in the figure below, which is significantly different from your result. Therefore, the trained result is significantly different from your result. For example, the difference between A2W and A2D is about 3 points, while the other 4 tasks are about 8 points apart. Attached is my code link. Could you please show me where the problem lies.
Github code link with attached code
https://github.com/dunanyang/AnyDA_Warmup.git
![image](https://github.com/omchakrabarty/omchakrabarty.github.io/assets/40531260/a5965886-3454-48de-9b96-a83f17bdb1e2)
![image](https://github.com/omchakrabarty/omchakrabarty.github.io/assets/40531260/e2ff1132-21bd-4013-92b1-a337a845f44d)

**Comment by dunanyang** [comment]
I have also sent you an email, and if you can receive it, I would be happy to ask for your advice through email.😊

**Comment by dunanyang** [comment]
I think the problem is quite complex. I may need a copy of the code or hyperparameter configuration. I look forward to hearing from you. My email is dny22@mails.tsinghua.edu.cn. Thank you very much!😊

**Comment by dunanyang** [comment]
I have looked at the data in DSBN, as shown in the figure below. For example, the Acc trained on Source Only from Amazon to Webcam is 73.5, mine is 75 to 78, and yours is 87. I would like to know how you implemented it. Thank you very much.
![image](https://github.com/omchakrabarty/omchakrabarty.github.io/assets/40531260/3315dfd1-7814-4060-8618-96331e602d06)

**Comment by dunanyang** [comment]
I am very interested in your work and would like to reproduce your results. I need your help🚀

**Comment by dunanyang** [comment]
Hello, I haven't received your email yet, so I would like to contact you again. Could you please answer the above question?

**Comment by omchakrabarty** [comment]
Hello again!
The code already in Github on which we had the discussion would suffice. Did you do the source-only pretraining as we discussed? Also, did you do the imagenet pretraining on the network before doing the source-pretraining? So as given in the paper, there are 3 steps -- **Imagenet pretraining --> Source-only training --> Full AnyDa training**. Are you following the exact protocol? What was your imagenet accuracy?
Also please NOTE, I just finished my graduation from IIT Kharagpur, India and will be travelling home in a few days. Sorry for your delayed replies. I have replied to your mail. Am closing the issue here. Kindly mail me directly. Will reply after travelling and reaching home

**Comment by omchakrabarty** [comment]
Broad discussions with necessary details are already mentioned. For more brainstorming, kindly mail to: omchakrabrarty@gmail.com
Also, I have just gradutated from IIT Kharagpur and will be leaving the university to travel back home. Sorry for the late replies.
